src/test/java/uk/gov/hmcts/reform/sendletter/services/LetterChecksumGeneratorTest.java

```diff
@@ -13,12 +17,62 @@
     @Test
     public void should_return_same_md5_checksum_hex_for_same_letter_objects() {
 
+        Supplier<LetterRequest> letterSupplier =
+            () -> new LetterRequest(
+                singletonList(
+                    new Document(
+                        "cmc-template",
+                        ImmutableMap.of(
+                            "key11", "value11",
+                            "key21", "value21"
+                        )
+                    )
+                ),
+                "print-job-1234",
+                ImmutableMap.of(
+                    "doc_type", "my doc type",
+                    "caseId", "123"
+                )
+            );
+
+        LetterRequest letter1 = letterSupplier.get();
+        LetterRequest letter2 = letterSupplier.get();
+
+        assertThat(LetterChecksumGenerator.generateChecksum(letter1))
+            .isEqualTo(LetterChecksumGenerator.generateChecksum(letter2));
+    }
+
+    @Test
+    public void should_return_same_md5_checksum_hex_for_same_letter_with_pdfs_objects() {
+
+        Supplier<LetterWithPdfsRequest> letterSupplier =
+            () -> new LetterWithPdfsRequest(
+                asList(
+                    "foo", "bar"
+                ),
+                "print-job-1234",
+                ImmutableMap.of(
+                    "doc_type", "my doc type",
+                    "caseId", "123"
+                )
+            );
+
+        LetterWithPdfsRequest letter1 = letterSupplier.get();
+        LetterWithPdfsRequest letter2 = letterSupplier.get();
+
+        assertThat(LetterChecksumGenerator.generateChecksum(letter1))
+            .isEqualTo(LetterChecksumGenerator.generateChecksum(letter2));
+    }
+
+    @Test
+    public void should_return_different_md5_checksum_hex_for_different_letter_objects() {
+
         LetterRequest letter1 = new LetterRequest(
             singletonList(new Document(
                 "cmc-template",
                 ImmutableMap.of(
                     "key11", "value11",
-                    "key21", "value21"
+                    "key12", "value12"
```

Comment:
github not doing the best job displaying diff. this is now in a different method...